ehrapy/preprocessing/_normalization.py

```diff
@@ -43,53 +52,57 @@ def _scale_func_group(
 
     edata = _prep_edata_norm(edata, copy)
 
-    if layer is None:
-        var_values = edata[:, vars].X.copy()
-    else:
-        var_values = edata[:, vars].layers[layer].copy()
+    var_indices = _get_var_indices(edata, vars)
+    X = edata.X if layer is None else edata.layers[layer]
+
+    if np.issubdtype(X.dtype, np.integer):
+        X = X.astype(np.float64)
```

Comment:
Why are we casting X to a specific type? This is not obvious behavior, is it?

Reply:
Ah I see. I think it's very unlikely that users will ever have an integer array in the first place but then I guess it's fine. float64 is kind of expensive though and I wonder whether we can getaway with float32 instead?